xen/include/public/io/displif.h

```diff
@@ -0,0 +1,611 @@
+ * o Allocate and initialize
+ *   event channels and buffers
+ * o Publish transport parameters
+ *   that will be in effect during
+ *   this connection.
+ *              |
+ *              |
+ *              V
+ * XenbusStateInitialised
+ *
+ *                                       o Query frontend transport parameters.
+ *                                       o Connect to the event channels.
+ *                                                      |
+ *                                                      |
+ *                                                      V
+ *                                      XenbusStateConnected
+ *
+ *  o Create and initialize OS
+ *  virtual DISPL as per configuration.
+ *              |
+ *              |
+ *              V
+ * XenbusStateConnected
+ *
+ *                                      XenbusStateUnknown
+ *                                      XenbusStateClosed
+ *                                      XenbusStateClosing
+ * o Remove virtual sound device
+ * o Remove event channels
+ *              |
+ *              |
+ *              V
+ * XenbusStateClosed
+ *
+ */
+
+/*
+ * REQUEST CODES.
+ */
+#define XENDISPL_OP_DUMB_CREATE           0
+#define XENDISPL_OP_DUMB_DESTROY          1
+#define XENDISPL_OP_FB_CREATE             2
+#define XENDISPL_OP_FB_DESTROY            3
+#define XENDISPL_OP_SET_CONFIG            4
+#define XENDISPL_OP_PG_FLIP               5
+
+/*
+ * EVENT CODES.
+ */
+#define XENDISPL_EVT_PG_FLIP              0
+
+/*
+ * XENSTORE FIELD AND PATH NAME STRINGS, HELPERS.
+ */
+#define XENDISPL_DRIVER_NAME              "vdispl"
+
+#define XENDISPL_RESOLUTION_SEPARATOR     "x"
+/* Field names */
+#define XENDISPL_FIELD_CTRL_RING_REF      "ctrl-ring-ref"
+#define XENDISPL_FIELD_CTRL_CHANNEL       "ctrl-channel"
+#define XENDISPL_FIELD_EVT_RING_REF       "event-ring-ref"
+#define XENDISPL_FIELD_EVT_CHANNEL        "event-channel"
+#define XENDISPL_FIELD_RESOLUTION         "resolution"
+
+/*
+ * STATUS RETURN CODES.
+ */
+/* Operation is not supported */
+#define XENDISPL_RSP_NOTSUPP              (-2)
+/* Operation failed for some unspecified reason (e. g. -EIO) */
+#define XENDISPL_RSP_ERROR                (-1)
+/* Operation completed successfully */
+#define XENDISPL_RSP_OKAY                 0
+
+/* Path entries */
+#define XENDISPL_PATH_CONNECTOR           "connector"
+
+/*
+ * Description of the protocol between frontend and backend driver.
+ *
+ * The two halves of a Para-virtual sound driver communicates with
+ * each other using a shared page and an event channel.
+ * Shared page contains a ring with request/response packets.
+ *
+ *****************************************************************************
+ *                            Frontend to backend requests
+ *****************************************************************************
+ *
+ * All request packets have the same length (64 octets)
+ *
+ *
+ * Request dumb creation - request creation of a DISPL dumb buffer.
```

Comment:
Probably you need to explain somewhere in comments decisions on memory sharing and buffers (dumb, dma, tilers). Also you need to exlpain set of currently supported features (i.e. do you plan to add planes? renderers? etc.)